src/App.js

```diff
@@ -127,9 +132,17 @@ const returnLoadedWalletComponent = (Component, props, rest) => {
   } else {
     // If was closed and is loaded we need to redirect to locked screen
     if (wallet.wasClosed()) {
-      return <Redirect to={{pathname: '/locked/'}} />;
+      return <Redirect to={{ pathname: '/locked/' }} />;
     } else {
-      return returnDefaultComponent(Component, props);
+      if (rest.loadingAddresses) {
+        // If wallet is still loading addresses we redirect to the loading screen
```

Comment:
I still don't know how to do this check and show the loading screen without the need to do this `if` in each of the screens. The possibilities I had this was the best one I thought, how would I do it only to show when needed?

Remembering that there are some screens that even if it's loading the addresses, they shouldn't show the loading screen (like the '/server/' screen)

Reply:
I haven't tested this snippet, but I think something like this may work:

```javascript
const WalletLoadedRoutes = ({ match }) => {
    if (this.props.is_loading) {
        return <LoadingScreen />
    }
    return (
        <Switch>
            <Route exact path="/create_token" component={CreateToken} />
            // All other routes can be added here.
        </Switch>
    );
}
```

I did it based on this example https://reacttraining.com/react-router/web/example/basic.

Reply:
I've used a different approach to refactor this part of the code, check the last commit please.

What do you think? @msbrogli

Reply:
I think it is much better. Thanks!